oidc-controller/api/core/acapy/client.py

```diff
@@ -0,0 +1,133 @@
+import requests
+import json
+import logging
+from uuid import UUID
+
+from .models import WalletPublicDid, CreatePresentationResponse
+from ..config import settings
+
+_client = None
+logger = logging.getLogger(__name__)
+
+WALLET_DID_URI = "/wallet/did/public"
+CREATE_PRESENTATION_REQUEST_URL = "/present-proof/create-request"
+PRESENT_PROOF_RECORDS = "/present-proof/records"
+
+
+class AcapyClient:
+    wallet_id = settings.ACAPY_WALLET_ID
+    wallet_key = settings.ACAPY_WALLET_KEY
+    acapy_host = settings.ACAPY_ADMIN_URL
+    acapy_admin_api_key = settings.ACAPY_ADMIN_URL_API_KEY
+    service_endpoint = settings.ACAPY_PUBLIC_SERVICE_URL
+
+    wallet_token: str = None
+
+    def __init__(self):
+        if _client:
+            return _client
+        super().__init__()
+
+    def get_wallet_token(self):
+        logger.debug(f">>> get_wallet_token")
+        resp_raw = requests.post(
+            self.acapy_host + f"/multitenancy/wallet/{self.wallet_id}/token",
+            json={"wallet_key": self.wallet_key},
+            headers={settings.ACAPY_WEBHOOK_URL_API_KEY_NAME: self.acapy_admin_api_key},
+        )
+        assert (
+            resp_raw.status_code == 200
+        ), f"{resp_raw.status_code}::{resp_raw.content}"
+        resp = json.loads(resp_raw.content)
+        self.wallet_token = resp["token"]
+        logger.debug(f"<<< get_wallet_token")
+
+        return self.wallet_token
+
+    def create_presentation_request(
+        self, presentation_request_configuration: dict
+    ) -> CreatePresentationResponse:
+        logger.debug(f">>> create_presentation_request")
+        if not self.wallet_token:
```

Comment:
Note for the future: we will likely want to have the vc-authn agent run outside of an agency, with the agency agent being an option (but not mandatory). We can discuss further, but since the agent is read-only I feel like having a dedicated standalone service might be best.

Since this targets a tenanted wallet, we should add a `TODO` to handle different instances properly.

Reply:
Will add a dedicated issue/task for this